Bowling/Bowling/Frames.cs

```diff
@@ -0,0 +1,177 @@
+            placeCount++;
+
+            if (!score.Equals(RollTypes.StrikeMark))
+            {
+                score = bowlingMarks.Substring(placeCount, 1);
+                frame = scoreRolls.ScoreSecondRoll(frame, score);
+                placeCount++;
+            }
+
+            framesCollection.Add(frame);
+            return framesCollection;
+        }
+
+        private List<Frame> CreateFrameWithBonusRoll(int frameNumber, string bowlingMarks, int marksLength, List<Frame> framesCollection)
+        {
+            Frame frame = new Frame();
+            Rolls scoreRolls = new Rolls();
+            var rollsRemaining = marksLength - placeCount;
+            var roll1 = bowlingMarks.Substring(placeCount, 1);
+            var roll2 = bowlingMarks.Substring(placeCount + 1, 1);
+            var roll3 = "0";
+
+            if (rollsRemaining == 3)
+            {
+                roll3 = bowlingMarks.Substring(placeCount + 2, 1);
+            }
+
+            frame = scoreRolls.ScoreRollsWithBonus(frame, roll1, roll2, roll3);
+
+            framesCollection.Add(frame);
+            return framesCollection;
+        }
+
+        public List<Frame> UpdateFrameScoresForStrikesAndSpares(List<Frame> framesCollection)
+        {
+            Frame currentFrame;
+            Frame oneFrameAfterCurrentFrame;
+            Frame twoFramesAfterCurrentFrame;
+
+            for (var frameNumber = 0; frameNumber <= frameNumberTotal - 2; frameNumber++)
+            {
+                currentFrame = framesCollection[frameNumber];
+                oneFrameAfterCurrentFrame = framesCollection[frameNumber + 1];
+
+                if ((frameNumberTotal - 2) - frameNumber >= 2)
+                {
+                    twoFramesAfterCurrentFrame = framesCollection[frameNumber + 2];
+                }
+                else
+                {
+                    twoFramesAfterCurrentFrame = framesCollection[frameNumberTotal - 1];
+                }
+
+                switch (currentFrame.frameOutcome)
+                {
+                    case (RollTypes.Strike):
+                        if (oneFrameAfterCurrentFrame.frameOutcome == RollTypes.Strike)
+                        {
+                            currentFrame.frameScore = currentFrame.frameScore + oneFrameAfterCurrentFrame.firstRoll + twoFramesAfterCurrentFrame.firstRoll;
+                        }
+                        else
+                        {
+                            currentFrame.frameScore = currentFrame.frameScore + oneFrameAfterCurrentFrame.firstRoll + oneFrameAfterCurrentFrame.secondRoll;
+                        }
+                        break;
+
+                    case (RollTypes.Spare):
+                        currentFrame.frameScore = currentFrame.frameScore + oneFrameAfterCurrentFrame.firstRoll;
+                        break;
+                }
+            }
+
+            return framesCollection;
+        }
+
+        public string GetBowlingMarks()
```

Comment:
OK, how about this...since I'm renaming my Frames class to BowlingScore I will create a new class called UserInput to separate that part of the logic.  I would like to leave calculating the score here as I think it makes sense now.